contracts/core/mpl-rewards/v1/test/helper/Rewards.sol

```diff
@@ -0,0 +1,740 @@
+
+            // Make StakeLocker public and stake tokens
+            pat.openStakeLockerToPublic(        address(stakeLocker1));
+            fay.approve(address(bPool),         address(stakeLocker1), amt);
+            assertTrue(fay.try_stakeTo(         address(stakeLocker1), amt));
+        }
+
+        fay.increaseCustodyAllowance(address(custodian), custodyAmt);
+
+        assertEq(stakeToken.custodyAllowance(address(fay), address(custodian)), custodyAmt);  // Fay gives custody to custodian
+        assertEq(stakeToken.totalCustodyAllowance(address(fay)),                custodyAmt);  // Total custody allowance goes up
+
+        assertTrue(!custodian.try_transferByCustodian(address(stakeToken), address(fay), address(fox),     custodyAmt));  // P/SL:INVALID_RECEIVER
+        assertTrue(!custodian.try_transferByCustodian(address(stakeToken), address(fay), address(fay),              0));  // P/SL:INVALID_AMT
+        assertTrue(!custodian.try_transferByCustodian(address(stakeToken), address(fay), address(fay), custodyAmt + 1));  // P/SL:INSUF_ALLOWANCE
+        assertTrue( custodian.try_transferByCustodian(address(stakeToken), address(fay), address(fay),     custodyAmt));  // Able to transfer custody amount back
+
+        assertEq(stakeToken.custodyAllowance(address(fay), address(custodian)), 0);  // Custodian allowance has been reduced
+        assertEq(stakeToken.totalCustodyAllowance(address(fay)),                0);  // Total custody allowance has been reduced, giving Fay access to funds again
+    }
+
+    /****************************/
+    /*** LP functions testing ***/
+    /****************************/
+    function stake_test(bool isPfdtStakeToken, uint256 amt, uint256 stakeAmt, IStakeToken stakeToken) public {
+        uint256 start = block.timestamp;
+
+        if (isPfdtStakeToken) {
+            mintFundsAndDepositIntoPool(fay, pool1, amt * USD, amt * USD);
+            pat.setLockupPeriod(address(pool1), 0);
+        } else {
+            setUpForStakeLocker(amt, sam, fay);
+        }
+
+        checkDepositOrStakeDate(isPfdtStakeToken, start, stakeToken, fay);
+
+        amt *= WAD;
+
+        assertEq(stakeToken.balanceOf(address(fay)),          amt);
+        assertEq(mplRewards.balanceOf(address(fay)),            0);
+        assertEq(mplRewards.totalSupply(),                      0);
+
+        hevm.warp(start + 1 days);  // Warp to ensure no effect on depositDates
+
+        assertTrue(!fay.try_stake(stakeAmt * WAD));  // Can't stake before approval
+
+        fay.increaseCustodyAllowance(address(mplRewards), stakeAmt * WAD);
+
+        assertTrue(!fay.try_stake(0));          // Can't stake zero
+        assertTrue( fay.try_stake(stakeAmt * WAD));  // Can stake after approval
+
+        assertEq(stakeToken.balanceOf(address(fay)),     amt);  // PoolFDT balance doesn't change
+        assertEq(mplRewards.balanceOf(address(fay)),     100 * WAD);
+        assertEq(mplRewards.totalSupply(),               100 * WAD);
+
```

Comment:
Why is this 100 * WAD and not `stakeAmt`?

Reply:
We should change this to `stakeAmt` and change `test_stake` in the other files to be fuzz tests

Reply:
Also remove extra spaces here

Reply:
It wasn't a fuzz test before so I did not choose to create a new one, If we think it is good to make a fuzz test then it is doable.